framework/src/base/Adaptivity.C

```diff
@@ -133,7 +135,7 @@ bool
 Adaptivity::adaptMesh()
 {
   bool meshChanged = false;
-  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time))
+  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time) && _step % _interval == 0)
```

Comment:
Hmm, I see. Then maybe we should just _ask_ the adaptivity object if it wants to be run. I.e. move the conditional into an interface method.